### ISSUE TITLE
BooleanInput onChange return boolean instead of event

### ISSUE THREAD

**Issue by Madebymaurice** [marmelab/react-admin#4273]
**What you were expecting:**

Before on the `onChange` function we've got the event, but now we receive the value of the boolean input

**Steps to reproduce:**

When you click on a post, then "Miscelaneous" tab and "Commentable" switch input, you can see in the console that you get a boolean instead of the event.

**Related code:**

https://codesandbox.io/s/happy-sound-ydf99?fontsize=14&hidenavigation=1&theme=dark

**Other information:**

Seems that the new `handleChange` only pass value to react-final-form but the `event` is not transmitted

**Environment**

* React-admin version: 3.1.2
* Last version that did not exhibit the issue (if applicable): 2.9.3
* React version:16.12
* Browser: Chrome or Firefox
* Stack trace (in case of a JS error):

**Comment by Luwangel** [comment]
Hi,

You're right about the fact that the onChange function passes only the state of the boolean input in react-admin v3. But it was also the case in v2.

In v2.9.3, file `react-admin/packages/ra-ui-materialui/src/input/BooleanInput.js`

``` js
    handleChange = (event, value) => {
        this.props.input.onChange(value);
    };
```

In v3.1.2, file `react-admin/packages/ra-ui-materialui/src/input/BooleanInput.tsx`

``` js
    const handleChange = useCallback(
        (event, value) => {
            finalFormOnChange(value);
        },
        [finalFormOnChange]
    );
```

And so I think it's not a bug. But maybe I miss something?

Regards,
Adrien

**Comment by Madebymaurice** [comment]
Yes i miss this, but why the event is now no returned ?

**Comment by Luwangel** [comment]
The <BooleanInput/> is currently not designed to return the source event but it returns the changed value.

We couldn't change that without introducing a breaking change, but it could be a great improvement. 

So I put the tag `help wanted` and we'll continue the discussion with the other core maintainers.

**Comment by Madebymaurice** [comment]
Hello,

For me the breaking change is that before we have access to the event in the `onChange` function like `Material-UI` did and now we can't.
But anyway thanks for your help, i'll follow the discussion and try to encapsulate the `Switch` input from `Material-UI` with the `Field` for connect it with my form.
Thanks

**Comment by WiXSL** [comment]
I think version 4 is suitable for having this change.

**Comment by WiXSL** [comment]
Implemented in #7202
